### PR TITLE
UCT/IB: Fix cq creation failure using legacy ibv api

### PR DESCRIPTION
**Pull request by dsharma283**
With the rdma-core version 28.0 or higher the error code
returned by libibverbs for the case where ibv_cq_ex_to_cq
is not supported by the underlying transport has changed.
The transport layer now returns EOPNOTSUPP instead of 
ENOSYS.
Ucx is broken for all the transport devices which do not
have ibv_cq_ex_to_cq implemented. Fixing the error code
check in uct for IB devices.

Signed-off-by: Devesh Sharma <devesh.sharma@broadcom.com>

### CODE REVIEW

**Comment by brminich** [comment]
ok to test

**Comment by brminich** [comment]
@dsharma283, can you pleasse fix commit title according to https://github.com/openucx/ucx/wiki/Guidance-for-contributors#general-guidelines?
(seems need to use upper case for `UCT`)

**Comment by dsharma283** [comment]
> 
> 
> @dsharma283, can you pleasse fix commit title according to https://github.com/openucx/ucx/wiki/Guidance-for-contributors#general-guidelines?
> (seems need to use upper case for `UCT`)

Updated the patch subject line. Added one more change to make it backwards compatible.

**Comment by shamisp** [comment]
@dsharma283 - Thanks for the patch ! Please add Broadcom copyright to the modified files. In order to merge the PR we need signed CLA: https://www.openucx.org/wp-content/uploads/2019/05/ucx-corporate-contributor-agrement.pdf  . Once it is signed, please send the copy to me. Thanks !

**Review by shamisp — CHANGES_REQUESTED** [review]
Pending - waiting for CLA

**Comment by dsharma283** [comment]
> @dsharma283 - Thanks for the patch ! Please add Broadcom copyright to the modified files. In order to merge the PR we need signed CLA: https://www.openucx.org/wp-content/uploads/2019/05/ucx-corporate-contributor-agrement.pdf . Once it is signed, please send the copy to me. Thanks !

Thanks Pasha for this note, let me take it to my Org and get back to you with approvals.

**Comment by dsharma283** [comment]
I am still taking time to come back on CLA. This may take some more time

**Comment by dsharma283** [comment]
@shamisp Hi, I have received a signed copy of CLA, Please let me know where I should send the Signed CLA. I will add the broadcom copyright to the modified files.

**Comment by shamisp** [comment]
Please email to admin at ucfconsortium.org . Once it is sent, please let me know and I will make sure that it is counter signed.

**Comment by dsharma283** [comment]
@shamisp Sent!

**Comment by shamisp** [comment]
The CLA was received.

**Review by shamisp — APPROVED** [review]

**Comment by shamisp** [comment]
@yosefe FYI

File: src/uct/ib/base/ib_iface.c
Comment (dmitrygx) [inline src/uct/ib/base/ib_iface.c:955]: pls, warp each condition by `()`:
```
if (!cq && ((errno == EOPNOTSUPP) || (errno == ENOSYS)))
```

**Review by dmitrygx — CHANGES_REQUESTED** [review]

**Comment by shamisp** [comment]
Please add Broadcom copyright to the header of the file.

File: src/uct/ib/base/ib_iface.c
Comment (dsharma283) [reply src/uct/ib/base/ib_iface.c]: Done!

**Comment by shamisp** [comment]
@yosefe I think this has to be back ported to 1.10.x ?

**Comment by yosefe** [comment]
> @yosefe I think this has to be back ported to 1.10.x ?

it's up to @dsharma283

**Comment by dsharma283** [comment]
> 
> 
> > @yosefe I think this has to be back ported to 1.10.x ?
> 
> it's up to @dsharma283

I can do that, it is good to have.

**Comment by shamisp** [comment]
thanks, please open PR for v1.10.x branch. Much appreciated.

**Comment by shamisp** [comment]
@dmitrygx Is it good to go? thanks !

**Review by dmitrygx — APPROVED** [review]

**Pull request merged**
